lib/ansible/plugins/action/synchronize.py

```diff
@@ -155,6 +155,11 @@ def run(self, tmp=None, task_vars=None):
         if task_vars is None:
             task_vars = dict()
 
+        # We make a copy of the args here because we may fail and be asked to
+        # retry. If that happens we don't want to pass the munged args through
+        # to our next invocation. Munged args are single use only.
+        self._task._tmp_args = self._task.args.copy()
```

Comment:
Its not an issue with the test, its an issue with the code.... All of this documented in the issue this fixes. https://github.com/ansible/ansible/issues/18281#issuecomment-261382356 specifically addresses this. I agree it should be fixed more generally, but that seems like a relatively massive change to ansible itself, and it was not the fix suggested to me. I doubt I would be able to work through the larger fix.